cocos/core/scene-graph/base-node.ts

```diff
@@ -137,9 +137,9 @@ export class BaseNode extends CCObject implements ISchedulable {
      * @readOnly
      */
     @editable
-    get children (): Node[] {
+    get children (): ReadonlyArray<Node[]> {
```

Comment:
```ReadonlyArray<Node>```

Reply:
Fixed, that's why I should never use github to modify my code directly 🤣